vms/evm/predicate/README.md

```diff
@@ -0,0 +1,141 @@
+
+
+# Predicate
+
+This package contains the predicate data structure and its encoding and helper functions to unpack/pack the data structure.
+
+## Encoding
+
+A byte slice of size N is encoded as:
+
+1. Slice of N bytes
+2. Delimiter byte `0xff`
+3. Appended 0s to the nearest multiple of 32 bytes
+
+
+## Results
+
+`predicate_results.go` defines how to encode `PredicateResults` within the block header's `Extra` data field.
+
+For more information on the motivation for encoding the results of predicate verification within a block, see [here](../../../vms/platformvm/warp/README.md#processing-historical-avalanche-interchain-messages).
+
+### Serialization
+
+Note: PredicateResults are encoded using the AvalancheGo codec, which serializes a map by serializing the length of the map as a uint32 and then serializes each key-value pair sequentially.
+
+PredicateResults:
+```
++---------------------+----------------------------------+-------------------+
+|             codecID :                           uint16 |           2 bytes |
++---------------------+----------------------------------+-------------------+
+|             results :  map[[32]byte]TxPredicateResults | 4 + size(results) |
++---------------------+----------------------------------+-------------------+
+                                                         | 6 + size(results) |
+                                                         +-------------------+
+```
```

Comment:
1. Markdown natively supports tables ([ref](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-tables)) so we don't have to do this code-block formatting
2. Although implied it's not clear what the columns mean without a row to indicate the legend.
3. Do we need the final row?